WPMedia/fusion-news-theme-blocks
# Subscriptions blocks v2

**If you have not filled out the checklist below, the pr is not ready for review.**

## Description

Migrate Subscriptions blocks from v1 to v2

Note:
I was unable to grab the [getSignedInIdentity()](https://github.com/WPMedia/arc-themes-blocks/blob/arc-themes-release-version-2.1.1/blocks/identity-block/components/identity/index.jsx#L30) from useIdentity(). useIdentity only is returning Identity & isInitialized. Not sure why the function is not returned, I can see everything is ok on Identity-block. Do you mind taking a look into this?
And for this reason, I re-write the function [here](https://github.com/WPMedia/arc-themes-blocks/pull/1768/files#diff-0d753ed843e4d2e87f52816a851ca9785e793a7dd581aff72be0f68091b5c25aR10).

## Jira Ticket

https://arcpublishing.atlassian.net/browse/ASUB-7843
https://arcpublishing.atlassian.net/browse/ASUB-7842
https://arcpublishing.atlassian.net/browse/ASUB-7901 >> Minor change, reported on identity block for Signup 

## Acceptance Criteria

Checkout & offers working

## Test Steps

- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout Subscriptions_blocks_v2`
2. Include `@wpmedia/subscriptions-block` inside blocks.json (fusion repo)
3. Run fusion repo with linked blocks npx fusion start -f -l @wpmedia/subscriptions-block

I just added a payment provider to your testing org.
This block has a different behavior depending if the user is logged In or not. 

Pages needed:
Login page
checkout page. This page will contain the checkout component. `offerURL: “link to offer page”`
offer page.  This page will contain the offer component.

Step 1: The user is logged In and redirected to the offer page
Step 2: Go into the offer page
Step 3: Select an item
![Screenshot 2023-11-08 at 12 29 28](https://github.com/WPMedia/arc-themes-blocks/assets/54566275/d52e6530-3d17-4693-addb-1f4a7b4e55fe) 
`Note: The icon for each item should be a checkbox icon. We will change this once themes team completes this`
Step 4. On checkout, you will be able to see the contact Info, and the items into the cart.
Include the country: e.g. US, MX, 
`Note: The country or region should be a dropdown. We will change this once themes team completes this`
Click on checkout
![Screenshot 2023-11-08 at 14 48 36](https://github.com/WPMedia/arc-themes-blocks/assets/54566275/64f47bfa-9302-47aa-acf9-84bd94855007)
Step 5. Enter the payment info & click on purchase subscription
For now we have stripe intents as the payment provider. You can use 
cardNumber: 4242 4242 4242 4242
MM/YY: 12/24
cvc: 123
zip: 45610
![Screenshot 2023-11-08 at 14 57 01](https://github.com/WPMedia/arc-themes-blocks/assets/54566275/661a9607-44ec-4780-b11a-94c32bff1cf8)
You should be redirected to the success URL
You will be able to see the subscription and order on Subs app, by searching for your user.

Also, if you are an anonymous user, when you fall in offers and you select an item, you should be redirected to the login page, and once you are logged In you can continue with the checkout process

## Effect Of Changes

### Before

_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After

_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Manifest pr for new block:
- Feature pack pr for local development:
- How to deploy to news: https://arcpublishing.atlassian.net/wiki/spaces/TI/pages/3138682964/News+Theme+2.0+Migration+Development+Deployment+Notes
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
